app/models/pipeline_run_stage.rb

```diff
@@ -123,26 +127,21 @@ def due_for_aegea_check?
   end
 
   def update_job_status
-    if !id || !started? || succeeded? || failed?
-      # This is called only from PipelineRun.update_status and only when the guard above is false.
+    if !id || !started?
       Airbrake.notify("Invalid precondition for PipelineRunStage.update_job_status #{id} #{job_id} #{job_status}.")
       return
     end
-    if checked?
-      # This can happen due to overly-eager frequent calls from pipeline_monitor through PipelineRun.update_job_status.
```

Comment:
How are you dealing with this race after your change?

Reply:
This check originally existed to prevent a loader job from being enqueued twice. Enqueuing of loader jobs has now been moved to monitor_results. In the new version, result_status indicates whether a loader job is in the queue, and we refrain from enqueuing new jobs until that status has been changed.